### PR TITLE
JSON array should be accepted as a valid payload type in data binding scenario

### PR DESCRIPTION
**Pull request by arukshani**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/10658

### CODE REVIEW

**Comment by MaryamZi** [comment]
Not solely related to this, but shouldn't we only support records (and not objects)? IMO, by allowing objects, we are allowing something like a JSON --> object conversion.

**Comment by arukshani** [comment]
After discussing with @MaryamZi and @lankavitharana we have decided to consider ballerina object as an invalid data binding type. Only records, strings, json, xml, byte array and  record array are allowed in data binding scenarios.

File: stdlib/http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java [hunk 1]
```diff
@@ -255,6 +251,50 @@ private static BValue populateAndGetEntityBody(BMap<String, BValue> inRequest,
...
+     * Convert a json to the relevant record type.
+     *
+     * @param entityBodyType Represents entity body type
+     * @param bjson          Represents the json value that needs to be converted
+     * @return the relevant ballerina record or object
+     */
+    private static BValue getRecord(BType entityBodyType, BValue bjson) {
+        try {
+            return JSONUtils.convertJSONToStruct(bjson, (BStructureType) entityBodyType);
+        } catch (NullPointerException ex) {
```
Comment (MaryamZi) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java]: Again, not a change introduced by this PR, but shouldn't we handle other possible exceptions too? Not sure why we've handled (only) a NPE here?

File: stdlib/http/src/main/java/org/ballerinalang/net/http/SignatureParams.java [hunk 2]
```diff
@@ -75,9 +75,9 @@ private void validateEntityBodyParam(ParamDetail entityBodyParam) {
                     "' as param name, but found '" + entityBodyParam.getVarName() + "'");
         }
         int type = entityBodyParam.getVarType().getTag();
-        if (type == TypeTags.OBJECT_TYPE_TAG || type == TypeTags.RECORD_TYPE_TAG || type == TypeTags.JSON_TAG
+        if (type == TypeTags.RECORD_TYPE_TAG || type == TypeTags.JSON_TAG
                 || type == TypeTags.XML_TAG || type == TypeTags.STRING_TAG || (type == TypeTags.ARRAY_TAG
-                && ((BArrayType) entityBodyParam.getVarType()).getElementType().getTag() == TypeTags.BYTE_TAG)) {
+                && validArrayType(entityBodyParam))) {
```
Comment (MaryamZi) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/SignatureParams.java]: Shall we re-format these few lines?

File: stdlib/http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java (hunk 1 above)
Comment (arukshani) [reply stdlib/http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java]: Other exceptions are handled by the calling function.

**Pull request merged**